### ISSUE TITLE
Replace the index Map with a Set of paths

### ISSUE THREAD

**Issue by mwanji** [webjars/webjars-locator#67]
I started using the constructor that takes an index and realised that we actually want to tell the locator which WebJars to include, rather than how to search them. So instead of passing in a Map with the path and the reversed path, we could pass in a Set of paths and the locator could internally index them as it sees fit.

@alexo and @huntc debated this in #15 and I think this solution might be the best of both worlds (performance without unnecessary coupling). While this is a breaking change, client migration could be as simple as calling Map#values() on the Map and wrapping the result in a Set.

**Comment by jamesward** [comment]
Sounds good to me. I think we should be able to do this and maintain source compatibility.

**Comment by mwanji** [comment]
> maintain source compatibility

By keeping both methods? Or some other way? One reason I don't like the current method is that it implies how the webjars are searched. While technically we could change the search method, and simply discard the map's keys, it feels dishonest.

**Comment by jamesward** [comment]
Yeah. That is true. The current architecture is a bit of a mess. Can we add the method for now and then tackle a bigger overhaul / refactor later?

**Comment by mwanji** [comment]
I guess so. I'll try to do a PR for this soon.

**Comment by mwanji** [comment]
Does being able to specify the set of paths obviate the need for a configurable prefix, as discussed in #66 ? I think that would be nice, as otherwise there starts to be a combinatorial explosion of constructors. Also, if an application constructs the index itself, it shouldn't need a prefix to be automatically added.

For example, to find files in "META-INF/resources/myapp", a client could do:

```java
WebJarAssetLocator locator = new WebJarAssetLocator(Collections.singleton("META-INF/resources/myapp/app.js"));

assertEquals("META-INF/resources/myapp/app.js", locator.getFullPath("app.js"));
```
I think that's preferable to:

```java
new WebJarAssetLocator(Collections.singleton("app.js"), "META-INF/resources/myapp");
```

**Comment by jamesward** [comment]
Yeah, that sounds right.
